pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go

```diff
@@ -166,7 +172,18 @@ func (m *Manager) Start() error {
 		for {
 			select {
 			case isShuttingDown := <-events:
-				klog.V(1).InfoS("Shutdown manager detected new shutdown event, isNodeShuttingDownNow", "event", isShuttingDown)
+				var shutdownType string
+				if isShuttingDown {
+					shutdownType = "shutdown"
+				} else {
+					shutdownType = "cancelled"
+				}
+				klog.V(1).InfoS("Shutdown manager detected new shutdown event", "event", shutdownType)
+				if isShuttingDown {
+					m.recorder.Eventf(m.nodeRef, v1.EventTypeNormal, kubeletevents.NodeShutdown, "Shutdown manager detected new shutdown event at %v", time.Now().Format(time.RFC3339))
+				} else {
+					m.recorder.Eventf(m.nodeRef, v1.EventTypeNormal, kubeletevents.NodeShutdown, "Shutdown manager detected cancellation at %v", time.Now().Format(time.RFC3339))
```

Comment:
nit: consider to reword make it more clear what exactly is cancelled:

`Shutdown manager detected shutdown cancellation.`

Reply:
Fixed.